tests/js/core/widget/core/Spin/Spin.js

```diff
@@ -1,181 +1,352 @@
+			setup: initHTML,
+			teardown: function () {
+				tau.engine._clearBindings();
+			}
+		});
+
+		test("Default options Spin test", 17, function () {
+			var spin = document.getElementById("spin"),
+				spinWidget = tau.engine.instanceWidget(spin, "core.Spin"),
+				spinClasses = {
+					spin: "ui-spin",
+					sample: "sample-spin"
+				},
+				defaultoptions = {
+					min: 0,
+					max: 9,
+					moduloValue: "enabled",
+					shortPath: "enabled",
+					duration: 600,
+					direction: "up",
+					rollHeight: "custom", // "container" | "item" | "custom"
+					itemHeight: 38,
+					momentumLevel: 0, // 0 - one item on swipe
+					scaleFactor: 0.4,
+					moveFactor: 0.4,
+					loop: "enabled",
+					labels: [],
+					digits: 0, // 0 - doesn't complete by zeros,
+					dragTarget: "document"
+				};
+
+			ok(spin.classList.contains(spinClasses.spin),
+				"Spin classname of element is defined");
+			ok(spin.classList.contains(spinClasses.sample),
+				"Custom classname of element is defined");
+
+			equal(spinWidget.option("min"), defaultoptions.min, "Default option min of Spin is " + defaultoptions.min);
+			equal(spinWidget.option("max"), defaultoptions.max, "Default option max of Spin is " + defaultoptions.max);
+			equal(spinWidget.option("moduloValue"), defaultoptions.moduloValue,
+				"Default option moduloValue of Spin is " + defaultoptions.moduloValue);
+			equal(spinWidget.option("shortPath"), defaultoptions.shortPath,
+				"Default option shortPath of Spin is " + defaultoptions.shortPath);
+			equal(spinWidget.option("duration"), defaultoptions.duration,
+				"Default option duration of Spin is " + defaultoptions.duration);
+			equal(spinWidget.option("rollHeight"), defaultoptions.rollHeight,
+				"Default option rollHeight of Spin is " + defaultoptions.rollHeight);
+			equal(spinWidget.option("itemHeight"), defaultoptions.itemHeight,
+				"Default option itemHeight of Spin is " + defaultoptions.itemHeight);
+			equal(spinWidget.option("momentumLevel"), defaultoptions.momentumLevel,
+				"Default option momentumLevel of Spin is " + defaultoptions.momentumLevel);
+			equal(spinWidget.option("scaleFactor"), defaultoptions.scaleFactor,
+				"Default option scaleFactor of Spin is " + defaultoptions.scaleFactor);
+			equal(spinWidget.option("moveFactor"), defaultoptions.moveFactor,
+				"Default option moveFactor of Spin is " + defaultoptions.moveFactor);
+			equal(spinWidget.option("loop"), defaultoptions.loop,
+				"Default option loop of Spin is " + defaultoptions.loop);
+			equal(spinWidget.option("digits"), defaultoptions.digits,
+				"Default option digits of Spin is " + defaultoptions.digits);
+			equal(typeof spinWidget.option("dragTarget"), typeof defaultoptions.dragTarget,
+				"Default option dragTarget of Spin is " + typeof defaultoptions.dragTarget);
+			spinWidget.value(5);
+			equal(spinWidget.value(), 5,
+				"Default option value of Spin is " + 5);
+			equal(typeof spinWidget.option("labels"), typeof defaultoptions.labels,
```

Comment:
why not check if spinWidget.option("labels") is empty?
equal(spinWidget.option("labels"), defaultoptions.labels);

Reply:
This test checks type of data returned by `option("labels")`. Maybe test doesn't make sense but it is not my change. I moved this part of tests from mobile profile.
...
After thinking and checking in implementation, `option("labels")` should returns `array`. I'll change this test.